src/api/MarkerBasedAPI.js

```diff
@@ -0,0 +1,147 @@
+/**
+ * @flow
+ * @file class for Box marker based API's to inherit common functionality from
+ * @author Box
+ */
+import Base from './Base';
+
+type Params = {
+    marker: string,
+    limit: number,
+    params?: string
+};
+
+type Data = {
+    entries: Array<any>
+};
+
+class MarkerBasedApi extends Base {
+    /**
+     * @property {Data}
+     */
+    data: Data;
+
+    /**
+     * @property {string}
+     */
+    marker: string;
+
+    /**
+     * @property {number}
+     */
+    limit: number;
+
+    /**
+     * @property {Object}
+     */
+    params: ?Object;
+
+    /**
+     * @property {string}
+     */
+    id: string;
+
+    /**
+     * @property {boolean}
+     */
+    shouldFetchAll: boolean;
+
+    /**
+     * Gets query params for the API
+     *
+     * @return the query params object
+     */
+    getQueryParameters(marker: string, limit: number, params: ?Object): Object {
+        const queryParams: Params = {
+            marker,
+            limit
+        };
+
+        return Object.assign(queryParams, params);
```

Comment:
Probably just use spread operator like its done on line 89 to be consistent.
Also begs the question, if this function is really needed since merge would be a one liner.

Reply:
Just used the spread operator in markerGetRequest() instead